src/handlers/milestone_prs.rs

```diff
@@ -145,24 +145,34 @@ async fn milestone_cargo(
     // <https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-pull-requests-associated-with-a-commit>,
     // but it is a little awkward to use, only works on the default branch,
     // and this is a bit simpler/faster. However, it is sensitive to the
-    // specific messages generated by bors, and won't catch things merged
-    // without bors.
-    let merge_re = Regex::new("(?:Auto merge of|Merge pull request) #([0-9]+)").unwrap();
-
-    let pr_nums = commits.iter().filter_map(|commit| {
-        log::info!(
-            "getting PR number for cargo commit {} (len={})",
-            commit.sha,
-            commit.commit.message.len()
-        );
-        merge_re.captures(&commit.commit.message).map(|cap| {
-            cap.get(1)
-                .unwrap()
-                .as_str()
-                .parse::<u64>()
-                .expect("digits only")
-        })
-    });
+    // specific messages generated by bors or GitHub merge queue, and won't
+    // catch things merged beyond them.
+    let merge_re =
+        Regex::new(r"(?:Auto merge of|Merge pull request) #([0-9]+)|\(#([0-9]+)\)$").unwrap();
+
+    let pr_nums = commits
+        .iter()
+        .filter(|commit|
+            // Assumptions:
+            // * A merge commit always has two parent commits.
+            // * Cargo's PR never got merged as fast-forward / rebase / squash merge.
+            commit.parents.len() == 2)
+        .filter_map(|commit| {
+            log::info!(
+                "getting PR number for cargo commit {} (len={})",
+                commit.sha,
+                commit.commit.message.len()
+            );
+
+            merge_re.captures(&commit.commit.message).map(|cap| {
```

Comment:
I'm wondering, would it be a good idea to only look at the first line of the commit message? I'm wondering what would happen if there is a stray mention of another issue/PR in the body somewhere, like:

```
fix(fix): Migrate cargo script manifests across editions (#14864)

This is a multi-line description
that happens to mention an issue (#1234)
somewhere in its text.
```

Reply:
Looking at this closer, this regex doesn't seem to work with a multi-line commit message. For example:

```
Improve error handling when PathSource is relative (#14854)

When editing dependencies with cargo, if a relative PathSource is
supplied cargo panics with "both paths are absolute". This is the
opposite of what's actually wrong leading to confusion.

Instead, use the same error formatting we use in other diff_paths calls
and return that error instead of panicking.
```

This is not matching here, since the `$` requires the `(#14854)` to be at the end of the message (since regex does not have mutliline enabled by default).

Reply:
Hah! I overlooked that difference. Thanks for investigating I'll update to check the first line only then.

Reply:
Addressed in 7229b2a8a9b7b4b1dbd5b2c2c442d6535f101ad5!